src/aiida/engine/processes/ports.py

```diff
@@ -209,6 +216,15 @@ def __setitem__(self, key: str, port: ports.Port) -> None:
         if hasattr(port, 'non_db_explicitly_set') and not port.non_db_explicitly_set:  # type: ignore[attr-defined]
             port.non_db = self.non_db  # type: ignore[attr-defined]
 
+        # If the port is not metadata (signified by ``is_metadata`` and ``non_db`` being ``False`` if defined) and it
+        # does not already define a serializer, set the default serializer to ``to_aiida_type``.
+        if (
```

Comment:
Just for me to understand, will this new logic work for both CalcJobs and WorkChains?

Reply:
Yes. The logic is added to the `PortnameSpace` which is the generic class used by the `ProcessSpec` which is used by all processes

Reply:
I believe you mentioned that calcfunction already handle this separately. I am assuming this will not clash with that?

Reply:
It is essentially a no-op because it would try to set the same default again. I guess technically we could probably simplify the process function code to no longer explicitly add it, but I don't think it is worth the hassle.

Reply:
Cool, thanks. Maybe at least add a TODO comment near the calcfunction code to explain this.